### PR TITLE
Change TimeWindow mininum from 5 to 1 minute due to Autoscale backend…

### PR DESCRIPTION
**Pull request by PatrickTseng**
… allows trigger TimeWindow to be 1 minute.

## Description

Autoscale backend now allows trigger TimeWindow to be 1 min. Change the code so user can set time window be like "00:01:00"

## Checklist

- [ ] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

File: src/Monitor/Monitor.Test/SessionRecords/Microsoft.Azure.Commands.Insights.Test.ScenarioTests.AutoscaleTests/TestAddAzureRmAutoscaleSetting.json [hunk 1]
```diff
@@ -4,15 +4,15 @@
...
..."timeWindow\": \"PT1M\", \"timeAggregation\": \"Average\", \"operator\": \"GreaterThan\", \"threshold\": 10.0 }, \"scaleAction\": { \"direction\": \"Increase\", \"value\": \"1\", \"cooldown\": \"PT5M\", \"type\": \"ChangeCount\" } }, { \"metricTrigger\": { \"metricName\": \"Requests\", \"metricResourceUri\": \"/subscriptions/b67f7fec-69fc-4974-9099-a26bd6ffeda3/resourceGroups/TestingMetricsScaleSet/providers/Microsoft.Compute/virtualMachineScaleSets/testingsc\", \"timeGrain\": \"PT2M\", \"statistic\": \"Average\", \"timeWindow\": \"PT1M\", \"timeAggregation\": \"Average\", \"operator\": \"GreaterThan\", \"threshold\": 15.0 }, \"scaleAction\": { \"direction\": \"Decrease\", \"value\": \"2\", \"cooldown\": \"PT6M\", \"type\": \"ChangeCount\" } } ], \"recurrence\": { \"frequency\": \"Week\", \"schedule\": { \"timeZone\": \"UTC\", \"days\": [ \"1\" ], \"hours\": [ 5 ], \"minutes\": [ 15 ] } } } ], \"enabled\": true, \"name\": \"MySetting\", \"targetResourceUri\": \"/subscriptions/b67f7fec-69fc-4974-9099-a26bd6ffeda3/resourceGroups/TestingMetricsScaleSet/providers/Microsoft.Compute/virtualMachineScaleSets/testingsc\" }, \"name\": \"MySetting\", \"location\": \"East US\", \"tags\": {}}",
```
Comment (markcowl) [inline src/Monitor/Monitor.Test/SessionRecords/Microsoft.Azure.Commands.Insights.Test.ScenarioTests.AutoscaleTests/TestAddAzureRmAutoscaleSetting.json]: Did you update these files individually?  You need to re-record them if you make a change like this.

File: src/Monitor/Monitor.Test/Autoscale/NewAzureRmAutoscaleRuleTests.cs [hunk 2]
```diff
@@ -59,10 +59,11 @@ public void NewAutoscaleRuleCommandParametersProcessing()
             Cmdlet.TimeGrain = TimeSpan.FromMinutes(1);
             Cmdlet.ExecuteCmdlet();
 
-            Cmdlet.TimeWindow = TimeSpan.FromMinutes(4);
+            // TimeWindow minimum value is now 1 min. 
+            Cmdlet.TimeWindow = TimeSpan.FromMinutes(0);
```
Comment (markcowl) [inline src/Monitor/Monitor.Test/Autoscale/NewAzureRmAutoscaleRuleTests.cs]: ??

Comment (PatrickTseng) [reply src/Monitor/Monitor.Test/Autoscale/NewAzureRmAutoscaleRuleTests.cs]: Minimum TimeWindow is now 1 min.  If you pass over 0 min, we're expecting ArgOutOfRange thrown.

File: src/Monitor/Monitor.Test/SessionRecords/Microsoft.Azure.Commands.Insights.Test.ScenarioTests.AutoscaleTests/TestAddAzureRmAutoscaleSetting.json (hunk 1 above)
Comment (PatrickTseng) [reply src/Monitor/Monitor.Test/SessionRecords/Microsoft.Azure.Commands.Insights.Test.ScenarioTests.AutoscaleTests/TestAddAzureRmAutoscaleSetting.json]: I change manually. What else do I need to change ?

Comment (markcowl) [reply src/Monitor/Monitor.Test/SessionRecords/Microsoft.Azure.Commands.Insights.Test.ScenarioTests.AutoscaleTests/TestAddAzureRmAutoscaleSetting.json]: These tests need to be re-recorded.  You cannot edit these files manually

File: src/Monitor/Monitor/ChangeLog.md [hunk 3]
```diff
@@ -29,6 +29,9 @@
 * Use common alert schema enabled for the receivers. This is not applicable for SMS, Azure App push , ITSM and Voice recievers
 * Webhooks now supports Azure active directory authentication .
 
+## Version 1.3.1
```
Comment (markcowl) [inline src/Monitor/Monitor/ChangeLog.md]: Just add this into the 'Upcoming Release' section.

File: src/Monitor/Monitor/help/New-AzAutoscaleRule.md [hunk 4]
```diff
@@ -274,7 +274,7 @@ Aliases:
 
 Required: False
 Position: Named
-Default value: None
+Default value: 1 minute
```
Comment (markcowl) [inline src/Monitor/Monitor/help/New-AzAutoscaleRule.md]: What is the impact of this change?  What is the impact now if no value is set.  Whant to make sure this is not a breaking change in behavior.

File: src/Monitor/Monitor.Test/Autoscale/NewAzureRmAutoscaleRuleTests.cs (hunk 2 above)
Comment (gucalder) [reply src/Monitor/Monitor.Test/Autoscale/NewAzureRmAutoscaleRuleTests.cs]: In fact, passing 0 here is the same as not sending the argument (it is the default value).
This has to be changed to something like TimeSpan.FromSeconds(59). That would throw the ArgOutOfRangeException

File: src/Monitor/Monitor/help/New-AzAutoscaleRule.md (hunk 4 above)
Comment (gucalder) [reply src/Monitor/Monitor/help/New-AzAutoscaleRule.md]: This did not change the behavior of "without value" (or value == 0) Without value should have used the default (5 minutes before, now 1 minute)

File: src/Monitor/Monitor/ChangeLog.md (hunk 3 above)
Comment (gucalder) [reply src/Monitor/Monitor/ChangeLog.md]: +1

**Comment by wyunchi-ms** [comment]
Hi @PatrickTseng , this pr last for a long time. Do you still need this pr? If yes, could you please take a look at this issue? We may close this issue in a few days, if it's no longger needed.

**Comment by wyunchi-ms** [comment]
Hi @PatrickTseng , I'm closing this PR now. You can reopen it when you are ready.

**Comment by neothoms** [comment]
We still need to set TimeWindow of 1 minute for our Autoscales Rules. 
Please proceed with this PR. 
Please reopen it.

**Comment by VeryEarly** [comment]
Hi
@PatrickTseng @neothoms @gucalder
 It was mentioned in swagger:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L411
The range of time window
"Must be between 12 hours and 5 minutes."

Can you please confirm.

**Comment by VeryEarly** [comment]
- the minimum time window updated from 5 min to 1 min without updating the swagger spec: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L411

- Timewindow was defined as required in swagger spec: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/monitor/resource-manager/Microsoft.Insights/stable/2015-04-01/autoscale_API.json#L371
but is optional parameter in powershell cmdlet:
https://github.com/Azure/azure-powershell/blob/master/src/Monitor/Monitor/Autoscale/NewAzureRmAutoscaleRuleCommand.cs#L81
with a default value which is not recommended in powershell

- Test cases related to New-AzAutoscaleRule need to be re-recorded due to the default value mentioned above was changed, the test case relied on a certain VMSS resource: https://github.com/Azure/azure-powershell/blob/master/src/Monitor/Monitor.Test/ScenarioTests/AutoscaleTests.ps1#L22
which is not there anymore I believe, please update test cases and re-record them.

**Comment by msJinLei** [comment]
@PatrickTseng Can you resolve the conflicts?

**Comment by msJinLei** [comment]
@PatrickTseng We are going to close the PR since no update for a long time. In addition, master is changing a lot during the time you are working on your feature. It would be better for you cherry pick your commit above the latest master and create a new PR rather than use this one.

**Pull request closed without merging**